OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -502,6 +506,29 @@ public void ModifyDeathActorInit(Actor self, TypeDictionary init)
 			init.Add(new FacingInit(Facing));
 		}
 
+		void INotifyBecomingIdle.OnBecomingIdle(Actor self)
+		{
+			OnBecomingIdle(self);
+		}
+
+		protected virtual void OnBecomingIdle(Actor self)
```

Comment:
This introduces a one-tick stall where aircraft stop moving when transitioning from flying to circling.

A specific workaround that may work here is to RunActivity at the end, but I suspect a proper fix needs to be implemented deeper in the `INotifyBecomingIdle` code.

Reply:
Fixed it in `Actor` via `RunActivity`, I couldn't come up with anything else.